internal/generator/vector/output/syslog/syslog.go

```diff
@@ -40,15 +40,23 @@ const (
 	nodeProcIdRFC5424 = `._syslog.proc_id = to_string!(.systemd.t.PID || "-")`
 
 	// Default values for Syslog fields for application logType and infrastructure logType if source 'container'
-	containerAppName  = `._syslog.app_name = join!([.kubernetes.namespace_name, .kubernetes.pod_name, .kubernetes.container_name], "_")`
+	containerAppName = `._syslog.app_name, err = join([.kubernetes.namespace_name, .kubernetes.pod_name, .kubernetes.container_name], "_")
+if err != null {
```

Comment:
Seems like we should log the error here and spam the collector logs.  This may be the only way for users to understand they need to set the value to something or not prune these fields